tests/Feature/Booking/BookingTest.php

```diff
@@ -275,85 +280,142 @@ public function testInvalidBookingCantBeCreated()
-            'position_id' => $position,
+            'position_id' => $this->position->id,
             'from' => '2019-08-10 14:00:00',
             'to' => '2019-08-10 15:00:00',
         ]);
 
-        $booking2 = factory(Booking::class)->create([
-            'position_id' => $position,
-            'from' => '2019-08-10 19:00:00',
-            'to' => '2019-08-10 20:00:00',
-        ]);
-
         // Test 1: Doesn't allow to book inside of booking
         $this->graphQL("
           mutation {
-            createBooking(user_id: 1300001, position: {
-                connect: {$position}
-            }, from:\"2019-08-10 14:30:00\", to:\"2019-08-10 15:30:00\") {
+            createBooking(user_id: {$this->mockUserId}, position_id: {$this->position->id}, from:\"2019-08-10 14:30:00\", to:\"2019-08-10 15:30:00\") {
                 id
             }
           }")->assertJsonPath('errors.0.message', "Can't have overlapping bookings for the same position!");
+    }
+
+    /** @test */
+    public function testOverlappingBookingsCannotBeCreatedOverlappingOnEnd()
+    {
+        $this->bypassRatingChecks();
+
+        factory(Booking::class)->create([
+            'position_id' => $this->position->id,
+            'from' => '2019-08-10 19:00:00',
+            'to' => '2019-08-10 20:00:00',
+        ]);
 
         $this->graphQL("
           mutation {
-            updateBooking(id: {$booking2->id}, from:\"2019-08-10 14:30:00\", to:\"2019-08-10 15:30:00\") {
+            createBooking(user_id: {$this->mockUserId}, position_id: {$this->position->id}, from:\"2019-08-10 18:30:00\", to:\"2019-08-10 19:30:00\") {
                 id
             }
           }")->assertJsonPath('errors.0.message', "Can't have overlapping bookings for the same position!");
+    }
+
+    /** @test */
+    public function testAllowsOverlappingBookingsOnDifferentPositions()
+    {
+        $this->bypassRatingChecks();
+
+        $creationJsonFormat = ['data' => ['createBooking' => ['id']]];
+        // twr position to meet the default rating check in setupOverlappingTest.
+        $differentPosition = factory(Position::class)->create(['callsign' => 'EGFF_TWR']);
 
-        // Test 2: Doesn't allow to book over booking
         $this->graphQL("
           mutation {
-            createBooking(user_id: 1300001, position: {
-                connect: {$position}
-            }, from:\"2019-08-10 14:00:00\", to:\"2019-08-10 15:00:00\") {
+            createBooking(user_id: {$this->mockUserId}, position_id: {$this->position->id}, from:\"2019-08-10 18:30:00\", to:\"2019-08-10 19:30:00\") {
                 id
             }
-          }")->assertJsonPath('errors.0.message', "Can't have overlapping bookings for the same position!");
+          }")->assertJsonStructure($creationJsonFormat);
 
         $this->graphQL("
           mutation {
-            updateBooking(id: {$booking2->id}, from:\"2019-08-10 14:00:00\", to:\"2019-08-10 15:00:00\") {
```

Comment:
You have removed all the testing for the update poking method in this test class. We should probably still test this somewhere

Reply:
See line 422.

Reply:
That test doesn't assert the booking checks are run though (they actually aren't at the moment in this case because the schema still runs `@update`. Either need to mock the service to assert the function called or test the funcitonality

Reply:
Added more tests under BookingsUpdateTest